pkg/core/chain/chain.go

```diff
@@ -713,6 +727,16 @@ func (c *Chain) provideRoundResults(r rpcbus.Request) {
 	r.RespChan <- rpcbus.Response{*buf, nil}
 }
 
+func (c *Chain) provideSyncProgress(r rpcbus.Request) {
+	c.mu.RLock()
+	prevBlockHeight := c.prevBlock.Header.Height
+	c.mu.RUnlock()
+
+	progressPercentage := (float64(c.highestSeen) / float64(prevBlockHeight)) * 100
```

Comment:
We can have here a check for divide-by-zero panic. I think there are some edge cases this can be triggered.

Reply:
Good catch, I will add it.